streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -264,8 +267,12 @@ public void init(final ProcessorContext context,
 
         segments.openExisting(this.context, observedStreamTime);
 
+        this.positionCheckpointFile = new File(context.stateDir(), this.name() + ".position");
```

Comment:
The logic of reading checkpoint upon init is duplicated on each of the classes, could we consolidate them further into `StoreQueryUtils`, e.g. by just passing in the stateDIR/stateName in it?

Reply:
Thanks! @patrickstuedi consolidated the logic of actually reading the position checkpoints into StoreQueryUtils, as you suggested. Also, after some cleanup, we only have two instances of this now, in `AbstractRocksDBSegmentedBytesStore` and in `RocksDBStore`.